test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -753,6 +753,31 @@ await using (var conn2 = await OpenConnectionAsync())
         }
     }
 
+    [Test]
+    [IssueLink("https://github.com/npgsql/npgsql/issues/4603")]
+    public async Task Reload_types_keepalive_concurrent()
+    {
+        if (IsMultiplexing)
+            Assert.Ignore("Multiplexing doesn't support keepalive");
+
+        var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
+        {
+            KeepAlive = 1,
+        };
+        using var _ = CreateTempPool(csb, out var connString);
+
+        await using var conn = await OpenConnectionAsync(connString);
+
+        var startTimestamp = Stopwatch.GetTimestamp();
+        // Give a few seconds for a KeepAlive to possibly perform
+        while (GetElapsedTime(startTimestamp).TotalSeconds < 2)
```

Comment:
We don't usually do this kind of concurrency testing (unreliable/flaky) - we can also do the fix without the test (or if you think it's important we can merge with).

Reply:
It is unreliable in a sense that it doesn't guarantee to reproduce the problem. But I think it's still valuable, since it has the potential to warn us about an issue (and it shouldn't produce false positives).

Reply:
Also keep in mind test perf (as this tight-loops for a bit). But am OK with merging it.

Reply:
Yeah, we probably should take a look at how the tests are performing (perhaps after 7.0 release?). While it's relatively OK on linux (2 minutes), on windows it takes about 3.

Reply:
Opened #4665 to track this.